workspace/welcome-to-xod/111-internet/patch.xodp

```diff
@@ -0,0 +1,171 @@
+{
+  "comments": [
+    {
+      "content": "Now you got a remote analog gauge for the data that you send to the cloud. You can place it on your desktop and watch for the moisture on your greenhouse in the countryside.\n\nAlso, another device might turn on or off the water pump, depending on the same data. Moreover, you can send a status of a water pump to the cloud and add a few nodes here to indicate that the water pump is working now.",
+      "id": "SyoB37C3I",
+      "position": {
+        "units": "slots",
+        "x": 10,
+        "y": 4
+      },
+      "size": {
+        "height": 2,
+        "units": "slots",
+        "width": 8
+      }
+    },
+    {
+      "content": "# Internet On Hardware\n\nYou might be familiar with the `tethering-inet` node and [XOD Cloud](https://xod.io/docs/guide/getting-started-with-feeds/) from previous chapters of the tutorial. In this chapter, we'll make an analog gauge for the data that you've sent before.\nFor simplicity, we placed here nodes to update the value and check out how your servo will react for it.\n\n## Excercise\n\n1. Place `xod-cloud/feeds/cloud-number` node to get the latest numeric value from the feed.\n2. Bind the same feed name into a `FEED` input pin.\n3. Link it with `tethering-inet`, `servo`, and `clock` nodes.\n4. Bind a `PORT` to the `servo` node.\n5. Run the simulation and check out how your servo is reacting. Update the value using the `tweak-number` on the right side of the program.\n\nNote that in the tutorial, we name a feed \"moisture\". Usually, the moisture is measured as a percentage. So we expect values in a range of 0.00...1.00. If you have another range of values, you'll need a `map` node right before the `VAL` input of a `servo` node.",
```

Comment:
# Internet On Hardware

You might be familiar with the `tethering-inet` node and [XOD Cloud](https://xod.io/docs/guide/getting-started-with-feeds/) from previous chapters of the tutorial. In this chapter, we make an analog gauge for the data that you've sent before.
Here, we placed nodes to update the value and check out how your servo will react for it.

## Excercise

1. Place the `xod-cloud/feeds/cloud-number` node to get the latest numeric value from the feed.
2. Bind the feed name into a `FEED` input pin. The feed name should be the same as at the `cloud-number` node.
3. Link the `cloud-number` node with `tethering-inet`, `servo`, and `clock` nodes.
4. Bind a `PORT` to the `servo` node.
5. Simulate the patch and look at how your servo is reacting. Update the cloud-stored value using the `tweak-number` on the right side of the program.

Note that in the tutorial, we named a feed "moisture". Usually, the moisture is measured in percentages. So we expect values in a range from 0.00 to 1.00. If you use another range, place the `map` node right before the `VAL` input of a `servo` node.